### PR TITLE
SIF syntactic tests and semantic checker integration

### PR DESCRIPTION
**Pull request by eileenchoe**
#362:
- Added syntactic tests for SIF importer requested in issue #362 
- Added the following warnings and errors: SIF_FORMAT_WARNING, SIF_UNWEIGHTED_RELATIONSHIP_TYPE_ERROR, SIF_MISSING_DATA_ERROR, SIF_STRAY_DATA_ERROR
- Wrote unit tests in import-sif-tests.js for all newly added SIF syntactic tests (separated from the GraphML syntactic tests)
- Enabled full functionality of Warnings and Error Modal for importers (SIF and GraphML), and fixed bug regarding rerouting the importer to the correct file type from the Warnings/Error Modal

#386:
- Semantic checker integrated in SIF and GraphML importers. SIF and GraphML network objects refactored to include positiveWeights and negativeWeights array, which were needed in the semantic checker. Unit tests refactored accordingly.

### CODE REVIEW

**Comment by NAnguiano** [comment]
I'm not @dondi buuuuuttttttt....

I checked it out and everything looks good to me! 👍

File: server/controllers/importers/sif.js [hunk 1]
```diff
@@ -37,6 +81,15 @@ module.exports = function (sif) {
...
 
   var networkType = sifNetworkType(entries);
+  if (networkType.warnings !== null) {
+    warnings.push(networkType.warnings);
+  }
+
+  if (networkType.errors != null) {
+    for (i = 0; i < networkType.errors.length; i++) {
+      errors.push(networkType.errors[i]);
+    }
```
Comment (dondi) [inline server/controllers/importers/sif.js]: This block seems rewritable as a `forEach` call:

    networkType.errors.forEach(function (error) {
        errors.push(error);
    });

It’s a tiny bit more compact.

File: test/errors-adjacency-matrix-modifications.js [hunk 2]
```diff
@@ -163,7 +163,7 @@ describe('errors-adjacency-matrix-modifications', function () {
       test.noErrors('test-files/adjacency-matrix-modifications/missing-data-output.xlsx');
     })
   })
-  
+
```
Comment (dondi) [inline test/errors-adjacency-matrix-modifications.js]: I know that technically, semicolons are optional in JavaScript, but I suggest that we still try to insert them where they are supposed to appear (e.g., after function calls).

File: web-client/public/js/upload.js [hunk 3]
```diff
@@ -252,28 +276,38 @@ $(function () {
     var emptyRowCount = warnings.filter(function(x){return x.warningCode=="EMPTY_ROW"});
     var invalidNetworkSizeCount = warnings.filter(function(x){return x.warningCode=="INVALID_NETWORK_SIZE"});
     var extraneousDataCount = warnings.filter(function(x){return x.warningCode=="EXTRANEOUS_DATA"});
+    var edgesWithoutWeightsCount = warnings.filter(function(x){return x.warningCode=="EDGES_WITHOUT_WEIGHTS"});
+    var edgeDefaultNotDirectedCount = warnings.filter(function(x){return x.warningCode=="EDGE_DEFAULT_NOT_DIRECTED"});
+    var sifFormatWarningCount = warnings.filter(function(x){return x.warningCode=="SIF_FORMAT_WARNING"});
```
Comment (dondi) [inline web-client/public/js/upload.js]: Inconsistent spacing here.

File: web-client/public/js/upload.js [hunk 4]
```diff
@@ -252,28 +276,38 @@ $(function () {
...
       for (var i = 0; i < warningCount.length; i++) {
         if (warningCount.length <= 3) {
-            warningsString += warningCount[i].errorDescription + "<br><br>";
+          appendWarning(warningCount[i]);
         } else if (printed[index] < 3){
-            warningsString += warningCount[i].errorDescription + "<br><br>";
-            printed[index]++;
+          appendWarning(warningCount[i]);
+          printed[index]++;
         } else if (printed[index] = 3) {
```
Comment (dondi) [inline web-client/public/js/upload.js]: I think this is intended to be `} else if (printed[index] === 3) {` ?

File: web-client/public/js/upload.js (hunk 3 above)
Comment (eileenchoe) [inline web-client/public/js/upload.js]: Cool, thanks for the feedback! What do you mean by inconsistent spacing here?

Comment (dondi) [inline web-client/public/js/upload.js]: Oh, I meant that everything was very tight, unlike the rest of the code. Plus I am just now noticing that `==` is being used and not `===`.

I would have formatted the code as follows:

    var edgeDefaultNotDirectedCount = warnings.filter(function (x) { return x.warningCode === "EDGE_DEFAULT_NOT_DIRECTED"; });

**Comment by eileenchoe** [comment]
Completed revisions suggested by @dondi's feedback.

**Comment by NAnguiano** [comment]
SIF syntactic tests and the error checker seem to be working just fine! I uploaded some bad SIF files and got the errors expected. I also uploaded some good ones and had no errors at all and some beautiful graphs. Exported SIF files work perfectly.

Pull Request Checklist
- [x] All Tests Pass
- [x] All Loaded Demos look as Expected
- [x] Reload works as expected
- [x] Loading File with Error brings up error modal, and it looks as expected
- [x] Loading File with Warning brings up warnings modal
- [x] Viewport Option Changing works as expected
- [x] Viewport Size Changing works as expected
- [x] Hide/Show Weights works as expected
- [ ] Force Sliders change the number above them and have an effect on the graph 
Slider values do not change, however the force is being correctly applied to the graph. See issue #364.
- [ ] Locking/Unlocking/Resetting/Undoing the force graph parameters works as expected
Locking, unlocking, and reset function as expected. Undo reset does not. See issue #364.
- [x] Weighted graph loaded with the "default to black edges" option checked appears as expected
See issue #422 for some questions on noticed behavior on this.
- [x] Import works for both SIF and GraphML
- [x] Graph can be exported to SIF and GraphML and exports can be re-uploaded without error

Pull request checklist from #440. Going to merge this one now.

**Pull request merged**